mlflow/transformers/model_io.py

```diff
@@ -123,7 +128,40 @@ def _load_component(flavor_conf, name, local_path=None):
         return cls.from_pretrained(repo, revision=revision)
 
 
-def _load_model(model_name_or_path, flavor_conf, accelerate_conf, device, revision=None):
+# FOR DEMO ONLY.
+#
+# inspect the HF config file to find the correct path to load the model from.
+#
+# this only works if we save the model's remote code in the `code` directory,
+# but should be pretty doable. for example, we can check if the model relies on
+# external code by doing something like:
+#
+# `transformers.utils.HF_MODULES_CACHE in inspect.getfile(model.__class__)`
+#
+# the HF config file contains the mapping between normal transformers constructs
+# and the custom code that was defined, so it should be as close to a source
+# of truth as we can get. internally, transformers does something similar,
+# see `get_class_from_dynamic_module()`:
```

Comment:
This is a good idea. +1
The base constructs needed to define model objects (or any other Auto loader type of pipeline component) would need to be mapped in this fashion in order for a model to be loadable within transformers. 
I think this is a good approach. The element to validate with this design would be for a model definition that is supplied in some nested directory and having the root defined within code_paths to determine if the solution permits discovering the correct module paths. 
If this becomes completely untenable (complexity akin to https://github.com/huggingface/transformers/blob/b6404866cda2952942a39e424206aed0cd6aeb5c/src/transformers/dynamic_module_utils.py ) we can limit the scope of support to require that dependent class definitions fit within the confines of transformers' ability to load these definitions. Their stance on support of custom code is far more restrictive than that of code_paths, after all. 
If the more restrictive solution is all that we can do (which is fine, by the way), we'll just need to add an additional try catch to alert users of what is required by way of code structuring when defining a custom model class and to give a working example in the docs (your test code that you wrote for building the mosaicml/mpy-7b model is a good candidate :) )